### ISSUE TITLE
Add help text to fields

### ISSUE THREAD

**Issue by guayom** [netlify/netlify-cms#552]
This is a feature request.
Sometimes you need to give special instructions on how to fill a field. 
For example, preferred image dimensions, the format for a phone number or just some help text you want to give to the user, like "This field is not shown on the page, but it is required for this and that reason ... "
Generally this text is shown below the field label or below the input.

It could be declared as such
```
- {label: Body, name: body, widget: markdown, helptext: "Your help text here..."}
```

I've found that this is an extremely useful feature. I use a lot in Dato. Many users don't edit the site very often, and they forget how they should fill some of the forms. Help texts solve that problem.

**Comment by neutyp** [comment]
I like this idea— any ideas on how to visualize it? is there a screenshot of Dato you could share?

**Comment by erquhart** [comment]
@neutyp Dato takes the standard approach, light gray text under the field:

<img width="543" alt="screen shot 2017-12-22 at 5 31 19 pm" src="https://user-images.githubusercontent.com/2112202/34314045-01043bf6-e73e-11e7-8293-8cda091c55f9.png">

**Comment by tech4him1** [comment]
Any design considerations with having it above the field (right below the field name) vs below the field?

**Comment by erquhart** [comment]
That'd be tough for us considering that we're using dog ear style labels:

<img width="783" alt="screen shot 2018-01-01 at 4 16 08 pm" src="https://user-images.githubusercontent.com/2112202/34471165-84c595d8-ef0f-11e7-8114-17b89a9537c5.png">

**Comment by tech4him1** [comment]
Ah, yes, our type of labels would make that hard, so I think below would work just as well. Personally, I always like to have help text next to the label, because I see them as accomplishing the same purpose. They both explain to the user what to "put in the box".

**Comment by erquhart** [comment]
That's a great point. @neutyp?

**Comment by neutyp** [comment]
Yeah, I'm not sure, here. Putting below is certainly good and fine if we can ensure the field/input/widget is only one line tall, which will certainly not always be true. If it's taller than one line, it feels visually too separated from the context.

At this point the best course of action would be to mock up some experiments of it located in different locations with differently sized fields/inputs/widgets; then discuss here visually

**Comment by erquhart** [comment]
Great point on widget height, is say it can't go underneath considering that.

**Comment by speexy** [comment]
I personally like a help-icon that shows me more info on click or hover for a quick reminder.

**Comment by kyleoliveiro** [comment]
What needs to be done for us to move forward with this feature?

**Comment by neutyp** [comment]
@erquhart Yeah, it's better than having no helper text. I'd say put it in, below the field, then move if we think of a better idea later.

**Comment by gil--** [comment]
I'm willing to begin developing this under the field. Didn't realize it was requested as i had created https://github.com/netlify/netlify-cms/issues/1426

**Comment by tech4him1** [comment]
@gil-- Awesome, it's yours! Feel free to ask here or in [our Gitter](https://gitter.im/netlify/NetlifyCMS) if you  have any questions.

**Comment by gil--** [comment]
One last thought after reading through the discussion here, I could add a secondary option on all widgets called: notePlacement: top/bottom OR showNoteAbove: true/false where it's under the widget by default as per https://github.com/netlify/netlify-cms/issues/552#issuecomment-395568749 and the user is able to show above widget on a case-by-case basis using the secondary field option.

**Comment by erquhart** [comment]
Agreed, getting simple support in for text below the field would be ideal, then we can iterate from there.

**Comment by gil--** [comment]
I've made a PR for this feature request, https://github.com/netlify/netlify-cms/pull/1429 

Let me know what you all think?

**Comment by alexandernanberg** [comment]
This can be closed since #1429 was merged correct? 😄

**Comment by tomrutgers** [comment]
Correct! Thanks @alexandernanberg
